presto-parquet/src/main/java/io/prestosql/parquet/ParquetTimestampUtils.java

```diff
@@ -54,11 +57,29 @@ public static long getTimestampMillis(Binary timestampBinary)
         long timeOfDayNanos = Longs.fromBytes(bytes[7], bytes[6], bytes[5], bytes[4], bytes[3], bytes[2], bytes[1], bytes[0]);
         int julianDay = Ints.fromBytes(bytes[11], bytes[10], bytes[9], bytes[8]);
 
-        return julianDayToMillis(julianDay) + (timeOfDayNanos / NANOS_PER_MILLISECOND);
+        long epochSeconds = (julianDay - JULIAN_EPOCH_OFFSET_DAYS) * SECONDS_PER_DAY + timeOfDayNanos / NANOS_PER_SECOND;
+        return new DecodedTimestamp(epochSeconds, (int) (timeOfDayNanos % NANOS_PER_SECOND));
     }
 
-    private static long julianDayToMillis(int julianDay)
+    public static <T extends Comparable<T>> Domain createDomain(PrestoTimestampEncoder<T> prestoTimestampEncoder, Binary singleValue, boolean hasNullValue)
     {
-        return (julianDay - JULIAN_EPOCH_OFFSET_DAYS) * MILLIS_IN_DAY;
+        T value = prestoTimestampEncoder.getTimestamp(decode(singleValue));
+        return TupleDomainParquetPredicate.createDomain(
+                prestoTimestampEncoder.getType(),
+                hasNullValue,
+                new ParquetRangeStatistics<T>()
+                {
+                    @Override
+                    public T getMin()
+                    {
+                        return value;
+                    }
+
+                    @Override
+                    public T getMax()
+                    {
+                        return value;
+                    }
+                });
```

Comment:
```
return Domain.create(ValueSet.of(prestoTimestampEncoder.getType(), value), hasNullValue);
```

and remove  `Comparable` from method sig

Reply:
i understand this is resolved = fixed locally?

Reply:
also, once simplified, i suggest to inline, especially because the conversion is valid only for min=max case

if you inline, there is only one place to comment about this